core/src/main/scala/spire/math/poly/SpecialPolynomials.scala

```diff
@@ -2,14 +2,25 @@ package spire
 package math
 package poly
 
+import scala.collection.compat.immutable.LazyList
 
 import spire.algebra.{Eq, Field, Ring}
 import spire.math.Polynomial
 import spire.syntax.field._
 
 object SpecialPolynomials {
 
+  // Horner scheme polynomial generator lazy list
+  def hornerSchemeLazy[C: Ring: Eq: ClassTag](zero: Polynomial[C], one: Polynomial[C],
```

Comment:
I'm not fond of appending `Lazy` to all the names, but sadly there are no `LazyList` parameters to overload on. open to suggestions for a better naming scheme (could make a separate object with a different name for them? I guess? idk if that's any better)

Reply:
To be honest, I don't think we necessarily need to keep two versions of the same methods here. I don't think there are that many users of `SpecialPolynomials`.

Reply:
I wasn't sure how important binary compatibility between versions was. I'm happy to remove the `Stream` versions

Reply:
spire is currently in 0.x, so we don't make any guarantees :wink: (that is to say, the deprecation cycle for the other methods is fine, but in this particular object I don't see a problem).